### ISSUE TITLE
is Orientation.unlockAllOrientations() asynchronous?

### ISSUE THREAD

**Issue by hjh5696** [yamill/react-native-orientation#187]
I am trying to figure out how to set specific component's orientation to landscape while others' locked to portrait.
So I did Orientation.lockToPortrait() at componentDidMount of the entry component. It works all well. In the specific component I tried
`Orientation.unlockAllOrientations();
Orientation.lockToLandscape();`

**Comment by andrerfneves** [comment]
@hjh5696 are you trying to set a Component inside of a View as `landscape` while the View itself is `portrait`? Not sure what the issue is.

**Comment by hjh5696** [comment]
@andrerfneves I am trying to keep a specific component `landscape` while other components of app `portrait`.

**Comment by andrerfneves** [comment]
@hjh5696 this package only does not allow you to keep certain components in a page landscape and others portrait. This package works on a per-screen basis. Do you mean `screen` instead of `components` in your question?

**Comment by hjh5696** [comment]
@andrerfneves Yes, sorry for delay. Actually I am using Modal from react native. When user click a button, the Modal component is set to be visible and I want it to be landscape.
below is the code
`componentWillMount() {`
    `    Orientation.unlockAllOrientations();`
    `    window.setTimeout(() => {`
  `    Orientation.lockToLandscape();`
    `    }, 100);`
  `    }`

 `//some other implementations here`

  `render() {`
    `return (`
      `<Modal onRequestClose={onRequestClose}>`
        `<ChildComponent/>`
      `</Modal>`
   ` );`
 ` }`

**Comment by andrerfneves** [comment]
@hjh5696 I would `lockToPortrait` inside your main application entry point, that way you can just run `unlockAllOrientations` to allow for both, and `lockToLandscape` if you want to force it. I find that `componentWillMount` works better than `componentWillUnmount`
